### PR TITLE
Implement add foods to meal tables

### PR DESCRIPTION
**Pull request by tmikeschu**
@Robbie-Smith @neight-allen ready for review!

Now, when a user who has foods in the foods.html table checks and adds a food to a meal, that food is added to the meal table for that date.

All tests passing.

---

Add DiaryDate class
Add max container width
Update localStorage data flow

[Finishes #141619143]

### CODE REVIEW

**Comment by Robbie-Smith** [comment]
@tmikeschu Looks good. Let's merge after we get the changes I made to the test files from my PR merged into master.

File: lib/LocalStorageHelper.js [hunk 1]
```diff
@@ -1,8 +1,34 @@
...
-      foods = localStorage.setItem('foods', '[]');
+      localStorage.setItem('foods', '[]');
+    }
+  }
+
+  setDate() {
+    const diaryDate = localStorage.getItem(this.today);
+    if (diaryDate === null) {
+      localStorage.setItem(this.today, '{}');
+    }
```
Comment (neight-allen) [inline lib/LocalStorageHelper.js]: This is just a style thing, but I think I'd just set `this.today` to `{breakfastFoods:[], lunchFoods:[], dinnerFoods:[], snackFoods:[]}` and you could cut out lines 6-9 and the `setMealFoods` function

File: lib/DiaryMeals.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    const caloriesCell = newRow.insertCell(1);
+    const deleteCell = newRow.insertCell(2);
+    this.setClassAndInnerHTML(nameCell, 'food-name', name);
+    this.setClassAndInnerHTML(caloriesCell, 'food-calories', calories);
+    this.setClassAndInnerHTML(deleteCell, 'food-delete', '<button>-</button>');
+  }
+
+  setClassAndInnerHTML(element, klass, html) {
+    element.className = klass;
+    element.innerHTML = html;
```
Comment (neight-allen) [inline lib/DiaryMeals.js]: I think I mentioned this on a closed PR already, but I prefer `className` over `klass`. You see `klass` used in all kinds of contexts and languages where `class` is reserved, but I've never thought it was good naming.

**Comment by Robbie-Smith** [comment]
@tmikeschu It looks great.

**Pull request merged**